### PR TITLE
Cherry picks for June release

### PR DESCRIPTION
**Pull request by jnummelin**
**What this PR Includes**

Cheery picks from `main` into `release-1.21` to prepare for June release

### CODE REVIEW

**Comment by jnummelin** [comment]
Waiting to be able to cherry pick #973

**Comment by jnummelin** [comment]
Needs also https://github.com/k0sproject/k0s/pull/980

**Comment by kke** [comment]
As mentioned in https://github.com/k0sproject/k0sctl/pull/152 - the versioning for the beta1 prerelease seems wrong. Maybe fix that in rc1.

File: embedded-bins/etcd/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,25 @@
-ARG BUILDIMAGE=golang:1.13-alpine
+ARG BUILDIMAGE=golang:1.16-alpine
```
[the remark this answers is not in the document]
Comment (jnummelin) [reply embedded-bins/etcd/Dockerfile:1]: 👍

**Pull request merged**